### PR TITLE
Clarify objection period for lazy consensus in SKIP 1

### PR DESCRIPTION
**Pull request by lagru**
## Description

@scikit-image/core, I'd like to clarify our expectations around how long of an objection period we expect before a lazy consensus is assumed. The suggestion is based on what I feel is already established practice.

I don't think this omission or vagueness has been a big problem in the past, but clarifying and documenting it seems like a good thing. I remember that last year when I started working for EOSS and started more actively proposing decisions myself, this would have given me a bit more confidence. :) 

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

### CODE REVIEW

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:150]:
```suggestion
  consensus). Core developers are expected to wait one or two days for others
  to give their opinion on the pull request if they’re not confident others
  would agree.
```

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:155]:
```suggestion
  on the issue or pull request page (lazy consensus). In this case, core
  developers are expected to wait about a week for others to give their
  opinion on the pull request, if they’re not confident others would agree.
```

**Review by mkcor — COMMENTED** [review]
Thanks for addressing this, @lagru! Explicit is better than implicit. :wink: 
For the sake of clarity, I prefer keeping the original (verbose) wording though.

**Comment by soupault** [comment]
@lagru, thank you for working on this! I fully support the idea of making the decision process less fuzzy and more dynamic. However, 1-2 days feels a bit too short of a response period to any change, and 7 days - as a period to reflect on a considerable code-related PR and to prepare a critical review (should be sufficient for docs-related PRs, though). I can easily imagine a situation where a core developer does not have any spare time to read through GitHub notifications and respond over the course of 1-2 days due to competing activities. In my opinion, 3 and 10(/14) days, respectively, would be less biased estimates for `“reasonable time”`. It would be great to hear more opinions!

File: doc/source/skips/1-governance.rst
Comment (stefanv) [reply doc/source/skips/1-governance.rst:150]: I like this suggestion; some of consensus operates on an understanding of what would cause conflict, and what would be fine to continue with.

Comment (lagru) [reply doc/source/skips/1-governance.rst]: See https://github.com/scikit-image/scikit-image/pull/7020/commits/e036dec4204f31f69933c6803f3f3580dfae86c4. Languagetool recommend a comma after "confident". :thinking:

**Comment by lagru** [comment]
> I can easily imagine a situation where a core developer does not have any spare time to read through GitHub notifications and respond over the course of 1-2 days due to competing activities. In my opinion, 3 and 10(/14) days, respectively, would be less biased estimates for “reasonable time”. It would be great to hear more opinions!

I totally get this. However, there's also a trade-off to increasing the objection period. The longer we wait, the easier the pending decision is forgotten again. At worst, a PR that is ready to merge becomes stale again and is re-discovered a few months later when everyone has forgotten the context. I really really want to avoid this case. 

Does it alleviate your concerns somewhat, that we can always revert a merged PR very quickly, if there are any concerns discovered after the fact? I just pushed https://github.com/scikit-image/scikit-image/pull/7020/commits/fe12d04e94e3499294fef18a4856f9d2029eebde, so that there is an even longer grace period for serious changes that require a SKIP.

Though, I get your case. It feels like a trade-off for which the right balance is probably entirely subjective. Right now, I would love to know the response time distribution between
- comments on a pull request and issues
- pull request and issue creation to the first response by each unique participant

to make a more informed decision here. :genie: Maybe I should wip up a quick notebook. :thinking:

File: doc/source/skips/1-governance.rst
Comment (mkcor) [reply doc/source/skips/1-governance.rst]: > Languagetool recommend a comma after "confident"

Hm, maybe because really it stands for "if they’re not confident *that* others would agree."

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:149]:
```suggestion
  consensus). Core developers are expected to wait three or four days for others to give
```

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:155]:
```suggestion
  expected to wait about two weeks for others to give their opinion if they’re not
```

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:156]:
```suggestion
  confident, that others would agree.
```

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:160]:
```suggestion
  decision-making process outlined above. But, in this case, the objection period
  should be one-month-long.
```

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:21]: Should we update to today's date?

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:150]: Do the two instances of "others" above mean "other core developers?"

File: doc/source/skips/1-governance.rst
Comment (mkcor) [reply doc/source/skips/1-governance.rst:154]: Not sure how to interpret "in case of such changes:"

* in case of such requests?
* in case the requested changes have been applied?

**Review by hmaarrfk — APPROVED** [review]

File: doc/source/skips/1-governance.rst
Comment (lagru) [reply doc/source/skips/1-governance.rst:154]: I think @soupault didn't feel that "in this case" was clear enough? I don't mind either version. To me both just reference the described changes above clearly enough.

Comment (lagru) [reply doc/source/skips/1-governance.rst:154]: I've never seen pull request written with a dash. If this is not a grammar thing I'd say "pull request" is the more consistent one with the docs. But feel welcome to apply the patch. :)

File: doc/source/skips/1-governance.rst
Comment (lagru) [reply doc/source/skips/1-governance.rst:150]: Yes. Though I imagine that in practice and most cases any concerns by non-core devs would also be treated with the same politeness.

File: doc/source/skips/1-governance.rst
Comment (mkcor) [reply doc/source/skips/1-governance.rst:154]: > I've never seen pull request written with a dash. If this is not a grammar thing I'd say "pull request" is the more consistent one with the docs. But feel welcome to apply the patch. :)

My suggestion (above) is to replace "pull-request" with "pull request" precisely. Grammar-wise, it's definitely "a pull request" (noun); in "pull request page" (above) though, the noun is "page" and "pull request" acts as an adjective, that's why it wouldn't be left-field to hyphenate it (look up "compound adjectives"). Still, I suggest that we don't (in this case). :wink:

Comment (mkcor) [reply doc/source/skips/1-governance.rst:154]: > I think @soupault didn't feel that "in this case" was clear enough? I don't mind either version. To me both just reference the described changes above clearly enough.

Ok, thanks. Then I would say "in the case of disagreement or requested changes" to be clear; I agree with @soupault ("in this case" could mean "agreement by *two* core developers *and* no disagreement or requested changes (...)" since it's a single sentence/chunk right before "in this case").

File: doc/source/skips/1-governance.rst
Comment (mkcor) [reply doc/source/skips/1-governance.rst:150]: Right. Then let's not have it mean "other core developers." Thanks. The sentence is not the most beautiful but I'll settle. Maybe "voice" instead of "give?"

Comment (lagru) [reply doc/source/skips/1-governance.rst:150]: > Then let's not have it mean "other core developers."

It's not clear to me, should I update it or are you happy with this?

Comment (mkcor) [reply doc/source/skips/1-governance.rst:150]: I'm happy as is. If we meant "other core developers," I would have suggested:
```diff
-others
+other core developers
```
(to be explicit/unambiguous).

**Review by mkcor — APPROVED** [review]

**Comment by mkcor** [comment]
Should we give this a month-long objection period, since the change affects a SKIP? There would be 3 weeks left, since we had 2 approvals 7 days ago. Counting down...

File: doc/source/skips/1-governance.rst
Comment (stefanv) [inline doc/source/skips/1-governance.rst:155]: "about a week" is confusing; perhaps rather "at least N days", with N being 5?

File: doc/source/skips/1-governance.rst
Comment (stefanv) [inline doc/source/skips/1-governance.rst:159]: A month seems very long. Do we need that much time?

Comment (mkcor) [reply doc/source/skips/1-governance.rst:159]: Apparently, we do! It's been way over a month since three core devs have approved this PR, I'd love to merge it now, but I admit I'd prefer having @stefanv's approval as well.

File: doc/source/skips/1-governance.rst
Comment (mkcor) [reply doc/source/skips/1-governance.rst:155]: I don't find it confusing personally, but I don't mind replacing it with something like "somewhere between 5 and 10 days." The problem I see with "at least N days" is that it provides only a lower bound. Then, the (implicit) upper bound becomes, in my understanding, by default, *6 months*, which is the time for the dormant bot to ping us back. I find it too long and unreasonable. I find it reasonable to wait "at least 5 days" (but not more than 10, 15, or 20). Let's make it clear again that this waiting period is understood *with no activity* -- if a fierce discussion is going on, obviously I'm not saying it should be interrupted after 10 days!

**Comment by stefanv** [comment]
Can we get this sorted during the next community call @lagru?

**Comment by lagru** [comment]
I think the next call is when we are in Seattle? Happy to discuss it. Though, it seems like we settled on a consensus mostly, so IMO we can just merge this if you can live with the clarifications as well @stefanv. I guess in practice it will only be relevant as a good reference to fall back on if decisions seem important or controversial.

File: doc/source/skips/1-governance.rst
Comment (stefanv) [reply doc/source/skips/1-governance.rst:159]: OK to go ahead with this PR as-is, as to not hold it up, but I do suggest we pull it back to two weeks to give the dev team license to move more quickly.

File: doc/source/skips/1-governance.rst
Comment (mkcor) [inline doc/source/skips/1-governance.rst:160]: Ok, I'm fine with either time periods; @stefanv we can merge the PR with this change:
```suggestion
  should be two-week-long.
```

File: doc/source/skips/1-governance.rst
Comment (mkcor) [reply doc/source/skips/1-governance.rst:155]: Ok.

File: doc/source/skips/1-governance.rst
Comment (lagru) [reply doc/source/skips/1-governance.rst:160]: That's a pretty substantial changes given that other's have already approved this. So I'd rather merge this as is for now and see what kind of experience we have with this. I imagine one-month is totally fine since I would start counting once the SPEC is going live / leaves draft mode.

File: doc/source/skips/1-governance.rst
Comment (lagru) [inline doc/source/skips/1-governance.rst:15]: Added you as an author as well @mkcor. :wink:

File: doc/source/skips/1-governance.rst
Comment (stefanv) [reply doc/source/skips/1-governance.rst:161]: OK, merging this for now; btw it should probably be "the objection period should be one month" or "one month long".

Comment (stefanv) [reply doc/source/skips/1-governance.rst:161]: Actually, going to let you fix that and then merge :)

Comment (mkcor) [reply doc/source/skips/1-governance.rst]: > That's a pretty substantial change given that other's have already approved this. So I'd rather merge this as is for now and see what kind of experience we have with this. I imagine one-month is totally fine since I would start counting once the SPEC is going live / leaves draft mode.

Ok.

Comment (mkcor) [reply doc/source/skips/1-governance.rst]: > btw it should probably be "the objection period should be one month" or "one month long".

:thinking: I guess either work although they're different:

* "the period is one month" is definitely valid;
* "one-month-long" has adjective value as in "a one-month-long period" so you can say "the period is one-month-long" (right?).

Comment (stefanv) [reply doc/source/skips/1-governance.rst]: I believe you can use "a one-month-long period" or "the period is one month long", but not "the period is one-month-long".

**Pull request merged**
